src/pages/en/guides/data-fetching.md

```diff
@@ -72,33 +104,70 @@ const weather = json.data;
 <p>Weather: {weather.getCityByName.weather.summary.description}</p>
 ```
 
-:::note
-Remember, all data in Astro components is fetched when a component is rendered.
-
-Your deployed Astro site will fetch data **once, at build time**. In dev, you will see data fetches on component refreshes. If you need to re-fetch data multiple times client-side, use a [framework component](/en/core-concepts/framework-components/) or a [client-side script](/en/core-concepts/astro-components/#client-side-scripts) in an Astro component.
-:::
+## Fetch from a Headless CMS
 
+Fetch remote content from your favorite CMS like StoryBlok or WordPress!
 
-## `fetch()` in Framework Components
+With an API key, some helper functions, and `getStaticPaths()` to generate dynamic page routes, an `.astro` component can fetch and render data from your CMS to generate your page content.
 
-The `fetch()` function is also globally available to any [framework components](/en/core-concepts/framework-components/):
+Here are some examples of what this data fetching looks like in Astro, with links to full tutorials below.
 
-```tsx title="src/components/Movies.tsx" /await fetch\\(.*?\\)/
-import type { FunctionalComponent } from 'preact';
-import { h } from 'preact';
-
-const data = await fetch('https://example.com/movies.json').then((response) =>
-  response.json()
-);
+### Example: StoryBlok API
 
-// Components that are build-time rendered also log to the CLI.
-// When rendered with a client:* directive, they also log to the browser console.
-console.log(data);
+```astro
+---
+// src/pages/[slug].astro
+// Fetch a list of your StoryBlok page links using @storyblok/js and axios
```

Comment:
As we don’t _show_ `axios` in use (it’s just internal to Storyblok’s API client), we probably shouldn’t mention it here. (I know I mentioned it, but I think it would only need to be included if we were showing install instructions.)

```suggestion
// Fetch a list of your StoryBlok page links using @storyblok/js
```